test/Interop/Cxx/templates/Inputs/class-template-non-type-parameter.h

```diff
@@ -0,0 +1,11 @@
+#ifndef TEST_INTEROP_CXX_TEMPLATES_INPUTS_CLASS_TEMPLATE_NON_TYPE_PARAMETER_H
+#define TEST_INTEROP_CXX_TEMPLATES_INPUTS_CLASS_TEMPLATE_NON_TYPE_PARAMETER_H
+
+template<class T, auto Size>
+struct MagicArray {
+    T t[Size];
+};
+
+typedef MagicArray<int, 2> MagicIntPair;
```

Comment:
What about a failing test to make sure we don't crash if the user tries to instantiate `MagicArray` or a similar type with a non-type template parameter that's defaulted?

Reply:
Great point! :) So I made MagicArray parameters defaulted and all works fine when they're explicitly mentioned, but not when we want to rely on the default. This could be improved, but I'd leave it for a separate PR as I expect quite a bit discussion over that.

Reply:
Sounds good to me.